php/php.editor/src/org/netbeans/modules/php/editor/completion/PHPCodeCompletion.java

```diff
@@ -1482,6 +1482,20 @@ public boolean isAccepted(PhpElement element) {
                             // XXX currently, only when mixins are used directly in the class. should support all cases?
                             accessibleTypeMembers.addAll(request.index.getAccessibleMixinTypeMembers(typeScope, enclosingType));
                         }
+                    } else if (typeScope instanceof EnumElement) {
+                        // add methods of BackedEnum/UnitEnum interface
+                        EnumElement enumElement = (EnumElement) typeScope;
+                        String enumInterfaceName = enumElement.getBackingType() != null ? "\\BackedEnum" : "\\UnitEnum"; // NOI18N
+                        final NameKind nameQuery = NameKind.exact(QualifiedName.create(enumInterfaceName));
+                        Set<InterfaceElement> enums = request.index.getInterfaces(nameQuery);
+                        for (InterfaceElement backedEnum : enums) {
+                            accessibleTypeMembers.addAll(request.index.getAccessibleTypeMembers(backedEnum, backedEnum));
+                        }
+                        if (enumElement.getBackingType() != null
+                                && !enumElement.getBackingType().toString().isEmpty()
+                                && !staticContext) {
```

Comment:
Nitpick: this variable could be checked earlier, to avoid calling `toString()` above at all.

_Note:_ this comment is - kind-of - in a conflict with my previous one :sweat_smile: